Makefile

```diff
@@ -347,6 +350,44 @@ bundle-run: operator-sdk ## Run bundle image
 bundle-cleanup: operator-sdk ## Remove bundle installed via bundle-run
 	$(OPERATOR_SDK) -n $(BUNDLE_RUN_NAMESPACE) cleanup $(OPERATOR_NAME)
 
+# Build a file-based catalog image
+# https://docs.openshift.com/container-platform/4.14/operators/admin/olm-managing-custom-catalogs.html#olm-managing-custom-catalogs-fb
+# NOTE: CATALOG_DIR and CATALOG_DOCKERFILE items won't be deleted in case of recipe's failure
+CATALOG_DIR := catalog_dir
+CATALOG_DOCKERFILE := ${CATALOG_DIR}.Dockerfile
+CATALOG_INDEX := $(CATALOG_DIR)/index.yaml
+
+.PHONY: catalog-build
+catalog-build: opm ## Build a file-based catalog image.
+	# Remove the catalog directory and Dockerfile if they exist
+	-rm -r ${CATALOG_DIR} ${CATALOG_DOCKERFILE}
```

Comment:
If you first remove the directory then why afterward you would need to delete a file that was in that directory?

Reply:
The file isn't inside the directory, it's prefixed with the directory's name
e.g.
`catalog_dir` and `catalog_dir.Dockerfile`

Reply:
Why? Is there a reason for that? 
It looks better to have `catalog.Dockerfile` instead of `catalog_dir.Dockerfile` similar to the naming format of the bundle Dockerfile that we currently have [`bundle.Dockerfile`](https://github.com/medik8s/machine-deletion-remediation/blob/main/bundle.Dockerfile)
WDYT?

BTW it would be nicer to have this file under the CATALOG_DIR directory if we want to have it as a temporary file.

Reply:
> Why? Is there a reason for that?

`opm generate dockerfile` works this way [1] :shrug: 

> It looks better to have catalog.Dockerfile

I can change the name of the `<catalog_dir>` no problem (of course we don't keep this file, but I don't have strong preferences)

[1] https://docs.openshift.com/container-platform/4.14/operators/admin/olm-managing-custom-catalogs.html

Reply:
>CATALOG_DIR := catalog

It is fine, and it would result in `catalog.Dockerfile` according to steps 1b and 1c in [OCP docs](https://docs.openshift.com/container-platform/4.15/operators/admin/olm-managing-custom-catalogs.html#olm-creating-fb-catalog-image_olm-managing-custom-catalogs).